src/db.php

```diff
@@ -280,6 +280,45 @@ public function getMSQ($id)
 		return $html;
 	}
 	
+	public function getMSQForDownload($id)
+	{
+
+		if (!$this->connect()) return null;
+		
+		$xml = FALSE;
+		
+		try
+		{
+			$st = $this->db->prepare("SELECT xml FROM msqs INNER JOIN metadata ON metadata.msq = msqs.id WHERE metadata.id = :id LIMIT 1");
+			DB::tryBind($st, ":id", $id);
+			$st->execute();
+			if ($st->rowCount() > 0)
+			{
+				$result = $st->fetch(PDO::FETCH_ASSOC);
+				$st->closeCursor();
+				$xml = $result['xml'];
+				if ($xml === NULL)
```

Comment:
`$xml` _should_ never bull `NULL`. The HTML is generated on-demand, and so needed these checks but this does not.